pkg/network/ebpf/c/runtime/conntrack-maps.h

```diff
@@ -0,0 +1,36 @@
+#ifndef __CONNTRACK_MAPS_H
+#define __CONNTRACK_MAPS_H
+
+#include "bpf_helpers.h"
+#include "tracer.h"
+#include "conntrack-types.h"
+
+/* This map is used for tracking conntrack entries
+ */
+struct bpf_map_def SEC("maps/conntrack") conntrack = {
+#ifdef BPF_F_NO_COMMON_LRU
+    .type = BPF_MAP_TYPE_LRU_HASH,
+#else
+    .type = BPF_MAP_TYPE_HASH,
+#endif
```

Comment:
The use of an LRU here is a major behavior change, and I'm worried about gating it on and off by kernel version since it adds another "mode" of operation to the system-probe.

If BPF_MAP_TYPE_HASH works for 4.4, can we use it everywhere ?

Reply:
The `BPF_MAP_TYPE_LRU_HASH` has better performance, and the LRU behavior doesn't kick in until the max number of entries is hit. This is actually better behavior than a hash, because it will evict old entries, which are most likely to be unnecessary, rather than erroring.

Reply:
These are the performance numbers running `samples/bpf/map_perf_test` from the kernel:
```
0:hash_map_perf pre-alloc 1526751 events per sec
3:hash_map_perf pre-alloc 1520085 events per sec
1:hash_map_perf pre-alloc 1364506 events per sec
2:hash_map_perf pre-alloc 1358230 events per sec
0:lru_hash_map_perf pre-alloc 2038742 events per sec
3:lru_hash_map_perf pre-alloc 1973669 events per sec
1:lru_hash_map_perf pre-alloc 2067042 events per sec
2:lru_hash_map_perf pre-alloc 2066950 events per sec
```

As you can see, the `BPF_MAP_TYPE_LRU_HASH` has better baseline performance.